From: Robert Schwebel <r.schwebel@pengutronix.de>
Subject: distutils need to care about cross compiling

If cross compiling it must be possible to overwrite the so_ext from the
outside.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

---
 Lib/distutils/command/build_ext.py |    2 ++
 1 file changed, 2 insertions(+)

Index: Python-3.0/Lib/distutils/command/build_ext.py
===================================================================
--- Python-3.0.orig/Lib/distutils/command/build_ext.py
+++ Python-3.0/Lib/distutils/command/build_ext.py
@@ -642,6 +642,8 @@ class build_ext(Command):
             ext_path[len(ext_path) - 1] = ext_path[len(ext_path) - 1][:8]
         # extensions in debug_mode are named 'module_d.pyd' under windows
         so_ext = get_config_var('SO')
+        if os.environ.get('CROSS_COMPILING') == 'yes':
+            so_ext = os.environ.get('SO')
         if os.name == 'nt' and self.debug:
             return os.path.join(*ext_path) + '_d' + so_ext
         return os.path.join(*ext_path) + so_ext
